### DISCUSSION TITLE
class visibility issues

### DISCUSSION THREAD

**Discussion by shuhuang-faire** [detekt/detekt#7787] [General]
I'm testing detekt 1.23.7 and notice some inconsistencies reqarding how the class visibility modifier is changed by autocorrect rules. For the following code
```
import misk.web.actions.WebAction

@Path("/foo")
class FooAction: WebAction
```
I used `val findings = autoCorrectRule.lint(code)` to lint it, it generates
```
import misk.web.actions.WebAction

@Path("/foo")
internal class FooAction: WebAction
```
as expected. Notice the *internal* modifier that was added by the lint(). 

But for the following:
```
import javax.ws.rs.GET

@Path("/foo")
class FooApi
```
after linting it is still the same (i.e. no *internal* added). 

This test works for 1.23.6 but it breaks after upgrading to 1.23.7. Is this an expected behavior? Thanks!

**Comment by cortinico** [comment]
> I used `val findings = autoCorrectRule.lint(code)` to lint it, it generates

Are you programmatically accessing the detekt API? Asking as it's a bit unclear to me what you're doing.

If what you're reporting is a valid bug, then also if you invoke `./gradlew detekt` with autoCorrection enabled you should notice extra `internal` modifiers (which is really weird).

Are you developing your own rule by any chance?

**Reply by shuhuang-faire** [reply]
Yes we are developing our own rules and the problem was caused by a Kotlin change that was not in the old Kotlin 1.9.23. We fixed it. Closing the issue.
Thanks!
